### PR TITLE
add e2e test to ensure all resources are in OpenAPI and work with `kubectl explain`

### PR DESCRIPTION
**Pull request by alexzielenski**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

This PR adds a test to enumerate all resources exposed by discovery and ensures they are exposed by OpenAPIV3 and work with `kubectl explain` for their long names, their specific api versions, their short names, and singular name, if available.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

/cc @jefftree @atiratree

### CODE REVIEW

File: test/e2e/kubectl/kubectl.go [hunk 1]
```diff
@@ -2064,6 +2065,66 @@ metadata:
...
 	})
+
+	ginkgo.Describe("kubectl explain", func() {
+		ginkgo.It("should be able to explain every resource advertised by the cluster", func() {
+			ginkgo.By("fetching all resources from cluster disccovery")
+			// Fetch k8s discovery. Ensure that every resource in discovery
+			// can be explained by kubectl explain
+			_, resources, err := f.ClientSet.Discovery().ServerGroupsAndResources()
+			framework.ExpectNoError(err, "expected to fetch discovery document successfully")
+
```
Comment (atiratree) [inline test/e2e/kubectl/kubectl.go:2076]: can we check that the resources returned are not empty?

File: test/e2e/kubectl/kubectl.go [hunk 2]
```diff
@@ -2064,6 +2065,66 @@ metadata:
...
+						for _, nameToTest := range namesToTest {
+							extraOpts := []string{}
+							if len(apiVersion) > 0 {
+								extraOpts = append(extraOpts, "--api-version", apiVersion)
+							}
+
+							args := append([]string{"explain", nameToTest}, extraOpts...)
+							ginkgo.By(fmt.Sprintf("calling `kubectl %s`", strings.Join(args, "")))
+							_, err := e2ekubectl.RunKubectl("", args...)
+							framework.ExpectNoError(err)
```
Comment (atiratree) [inline test/e2e/kubectl/kubectl.go:2119]: can we check that we get non empty output? We could at least  check for presence of the group, version and kind?

File: test/e2e/kubectl/kubectl.go [hunk 3]
```diff
@@ -2064,6 +2065,66 @@ metadata:
...
+
+					for _, apiVersion := range versionsToTest {
+						namesToTest := []string{gvr.Resource}
+						namesToTest = append(namesToTest, resource.ShortNames...)
+
+						// Ensure output contains correct Group Version and Kind
+						for _, nameToTest := range namesToTest {
+							extraOpts := []string{}
+							if len(apiVersion) > 0 {
+								extraOpts = append(extraOpts, "--api-version", apiVersion)
```
Comment (atiratree) [inline test/e2e/kubectl/kubectl.go:2113]: it would be good to run this also for openapi/v2 with `--output=plaintext-openapiv2`

**Comment by atiratree** [comment]
@alexzielenski we can now follow up with these tests since https://github.com/kubernetes/kubernetes/pull/118881 and https://github.com/kubernetes/kubernetes/pull/118879 have been merged

**Comment by Jefftree** [comment]
We can get this in after code freeze since it's just a test update

File: test/e2e/kubectl/kubectl.go [hunk 4]
```diff
@@ -2064,6 +2065,66 @@ metadata:
...
+			ginkgo.By("fetching all resources from cluster disccovery")
+			// Fetch k8s discovery. Ensure that every resource in discovery
+			// can be explained by kubectl explain
+			_, resources, err := f.ClientSet.Discovery().ServerGroupsAndResources()
+			framework.ExpectNoError(err, "expected to fetch discovery document successfully")
+
+			ginkgo.By("calling `kubectl explain` on each resource")
+
+			// These groups are not exposed by OpenAPI so can not be explained
+			excludeExpectedGroups := sets.New[string](
```
Comment (Jefftree) [inline test/e2e/kubectl/kubectl.go:2080]: Unnecessary now right?

**Comment by atiratree** [comment]
@alexzielenski can you please revisit this one?

**Pull request closed without merging**